jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/CustomRequestLog.java

```diff
@@ -318,11 +319,13 @@
 @ManagedObject("Custom format request log")
 public class CustomRequestLog extends ContainerLifeCycle implements RequestLog
 {
-    protected static final Logger LOG = LoggerFactory.getLogger(CustomRequestLog.class);
-
     public static final String DEFAULT_DATE_FORMAT = "dd/MMM/yyyy:HH:mm:ss ZZZ";
     public static final String NCSA_FORMAT = "%{client}a - %u %t \"%r\" %s %O";
     public static final String EXTENDED_NCSA_FORMAT = NCSA_FORMAT + " \"%{Referer}i\" \"%{User-Agent}i\"";
+    public static final String HANDLER_NAME = CustomRequestLog.class.getName() + ".handlerName";
+    public static final String REAL_PATH = CustomRequestLog.class.getName() + ".realPath";
+    public static final String USER_NAME = CustomRequestLog.class.getName() + ".userPrincipal";
```

Comment:
Make these an `enum` and then an EnumSet can be used to record which of them are required.